solidity/contracts/liquidity-protection/LiquidityProtectionSettings.sol

```diff
@@ -228,83 +262,90 @@ contract LiquidityProtectionSettings is ILiquidityProtectionSettings, AccessCont
+        external
+        onlyOwner()
+        validAddress(address(_poolAnchor))
+    {
+        emit NetworkTokenMintingLimitUpdated(_poolAnchor, networkTokenMintingLimits[_poolAnchor], _limit);
+
+        networkTokenMintingLimits[_poolAnchor] = _limit;
     }
 
     /**
-     * @dev returns high tier pools
+     * @dev increases the amount of network tokens minted into a specific pool
+     * can only be called by the minted tokens admin
      *
-     * @return high tier pools
+     * @param _poolAnchor   pool anchor
+     * @param _amount       amount to increase the minted tokens by
      */
-    function highTierPools() external view returns (address[] memory) {
-        uint256 length = _highTierPools.length();
-        address[] memory list = new address[](length);
-        for (uint256 i = 0; i < length; i++) {
-            list[i] = _highTierPools.at(i);
-        }
-        return list;
+    function incNetworkTokensMinted(IConverterAnchor _poolAnchor, uint256 _amount)
+        external
+        override
+        onlyMintedTokensAdmin
+        validAddress(address(_poolAnchor))
+    {
+        uint256 prevAmount = networkTokensMinted[_poolAnchor];
+        uint256 newAmount = prevAmount.add(_amount);
+        networkTokensMinted[_poolAnchor] = newAmount;
+
+        emit NetworkTokensMintedUpdated(_poolAnchor, prevAmount, newAmount);
     }
 
     /**
-     * @dev updates the system network token balance limits
-     * can only be called by the contract owner
+     * @dev decreases the amount of network tokens minted into a specific pool
+     * can only be called by the minted tokens admin
      *
-     * @param _maxSystemNetworkTokenAmount  maximum absolute balance in a pool
-     * @param _maxSystemNetworkTokenRatio   maximum balance out of the total balance in a pool (in PPM units)
+     * @param _poolAnchor   pool anchor
+     * @param _amount       amount to decrease the minted tokens by
      */
-    function setSystemNetworkTokenLimits(uint256 _maxSystemNetworkTokenAmount, uint32 _maxSystemNetworkTokenRatio)
+    function decNetworkTokensMinted(IConverterAnchor _poolAnchor, uint256 _amount)
         external
         override
-        onlyOwner()
-        validPortion(_maxSystemNetworkTokenRatio)
+        onlyMintedTokensAdmin
+        validAddress(address(_poolAnchor))
     {
-        emit SystemNetworkTokenLimitsUpdated(
-            maxSystemNetworkTokenAmount,
-            _maxSystemNetworkTokenAmount,
-            maxSystemNetworkTokenRatio,
-            _maxSystemNetworkTokenRatio
-        );
-
-        maxSystemNetworkTokenAmount = _maxSystemNetworkTokenAmount;
-        maxSystemNetworkTokenRatio = _maxSystemNetworkTokenRatio;
+        uint256 prevAmount = networkTokensMinted[_poolAnchor];
+
+        // allow the amount to reset to 0 if the provided amount is higher than the previous amount
+        uint256 newAmount = 0;
+        if (_amount < prevAmount) {
```

Comment:
Can this be the result of a valid flow?

Reply:
yeah, though it's more of a "correction" mechanism, for current protocol holdings and in also in case of a bug.

Reply:
Gotcha